src/boot/efi/boot.c

```diff
@@ -1880,18 +1931,12 @@ static BOOLEAN config_entry_add_loader_auto(
 
         /* do not add an entry for ourselves */
         if (loaded_image_path) {
-                UINTN len;
-                _cleanup_freepool_ CHAR8 *content = NULL;
-
-                if (StriCmp(loader, loaded_image_path) == 0)
+                assert(StriCmp(L"\\EFI\\BOOT\\BOOT" EFI_MACHINE_TYPE_NAME L".EFI", loader) == 0);
+                if (is_sd_boot(root_dir, loader, loaded_image_path))
+                        return FALSE;
+                /* Maybe the default loader is shim? */
+                if (is_sd_boot(root_dir, L"\\EFI\\BOOT\\GRUB" EFI_MACHINE_TYPE_NAME L".EFI", loaded_image_path))
```

Comment:
hmm, wha? grub?

Reply:
If the default loader (BOOTX64.EFI) is shim, then it will chaincall to GRUBX64.EFI after doing its thing. That may be us though (grub is just a name :P), and we don't want that.

Reply:
needs a comment